<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>RNA-seq background information, data analysis procedure, and details of the analysis tools &mdash; SPARTA-teaching 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="SPARTA-teaching 1.0 documentation" href="index.html" />
    <link rel="prev" title="Functionality wishlist" href="wishlist.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="wishlist.html" title="Functionality wishlist"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">SPARTA-teaching 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="rna-seq-background-information-data-analysis-procedure-and-details-of-the-analysis-tools">
<span id="rnaseqbackground"></span><h1>RNA-seq background information, data analysis procedure, and details of the analysis tools<a class="headerlink" href="#rna-seq-background-information-data-analysis-procedure-and-details-of-the-analysis-tools" title="Permalink to this headline">¶</a></h1>
<p>Before we dig into the data and begin trimming and aligning the reads to the genome, I think it is useful to understand what happens <em>after you submit your RNA to the sequencing facility</em>. This sort of knowledge can be very useful in understanding what could potentially provide bias and any number of issues to the end dataset. In this session we will cover several things including:</p>
<ol class="arabic simple">
<li><a class="reference internal" href="#rnaseqbackgroundinfo"><span>RNA-seq background information</span></a></li>
<li><a class="reference internal" href="#basicworkflow"><span>Basic analysis procedure</span></a></li>
<li><a class="reference internal" href="#trimmomatic"><span>Trimmomatic</span></a></li>
<li><a class="reference internal" href="#fastqc"><span>FastQC</span></a></li>
<li><span class="xref std std-ref">bowtie</span></li>
<li><span class="xref std std-ref">htseq</span></li>
<li><span class="xref std std-ref">edger</span></li>
</ol>
<div class="section" id="rna-seq-background-information">
<span id="rnaseqbackgroundinfo"></span><h2>RNA-seq background information<a class="headerlink" href="#rna-seq-background-information" title="Permalink to this headline">¶</a></h2>
<p>Before we begin, let&#8217;s watch a video about how <a class="reference external" href="https://www.youtube.com/watch?v=womKfikWlxM">Illumina sequencing works</a>.</p>
<p>This video does a pretty good job explaining how, in generalities the sequencing process works for DNA. So for sequencing RNA, the process is as follows:</p>
<img alt="RNAseq workflow" class="align-center" src="_images/rnaseqworkflow.jpg" />
<p><em>Adapted from: Zhernakova et al., PLoS Genetics 2013</em></p>
<p>So actually, we aren&#8217;t sequencing RNA at all! We are sequencing the cDNA made from the RNA. RNA-seq is a high resolution next generation sequencing (NGS) method to assess the transcriptome of an organism and compare transcriptional changes between organisms/treatments to ascertain specific pathways/genes that are moving in response. But now, let&#8217;s talk about what can add bias to the data and what we do with the data to make sure that it is reasonable to proceed to further analysis steps.</p>
<p>But first, let&#8217;s brainstorm a little bit. Look back at the RNA-seq workflow figure above and let&#8217;s suggest a few places where things could potentially affect the output dataset.</p>
<p>Here are a few thoughts...</p>
<blockquote>
<div><ul class="simple">
<li>How could the random priming step affect downstream results?</li>
<li>How could RNA secondary structures affect the library preparation process?</li>
<li>Would GC content be a problem?</li>
<li>Could gene length cause issues?</li>
<li>What might happen if you have genes with substantially different expression levels?</li>
<li>During the cluster generation on the Illumina flow cell, what might happen if you have too few clusters? Too many?</li>
<li>How is it possible to sequence many samples at one time?</li>
<li>What if you run out of reagents from one kit and have to open another kit to finish the library preparation process?</li>
<li>Could sequencing depth be an issue?</li>
</ul>
</div></blockquote>
<p>So now that you may be questioning the validity of any RNA-seq dataset, take heart! Many very smart people have thought about these issues and come up with ways to assess technical artifacts and correct for them. So again, let&#8217;s brainstorm some potential solutions to these problems. Which problems can be addressed through better chemistries/processes vs. mathematical/computational correction?</p>
<p>These sorts of issues should always be considered, but recognize that RNA-seq is becoming fairly commonplace and solutions to many of these questions exist. Be critical of your data and <em>always</em> look at the raw data.</p>
<p>Multiplexing the sequencing process by pooling several samples together is not only cheaper, it can overcome what are known as <em>batch effects</em>. Batch effects are when you have samples that correlate with one another based on batch/time/etc. instead of biological replication. This is a very real phenomenon and can be caused by using different lots of the same kit/flow cells when preparing samples! You can correct for this, but we will get there later... For now, have a look at the diagram showing how multiplexing is achieved.</p>
<img alt="Multiplexing samples diagram" class="align-center" src="_images/multiplex.jpg" />
<p><em>From: http://www.illumina.com/content/dam/illumina-marketing/documents/products/sequencing_introduction_microbiology.pdf</em></p>
<p>This is an example of what a <em>batch effect</em> looks like. Note how DMSO1 and ETZ1 group together and DMSO2 and ETZ2 group together (e.g. by batch).</p>
<img alt="Batch effect example" class="align-center" src="_images/batcheffect.jpg" />
<p>We can determine what is considered a &#8220;good&#8221; base call from a &#8220;bad&#8221; one through using what is known as the Phred scoring system or Q-score.</p>
<p>Where Q is defined as a property that is logarithmically related to the base call error probability:</p>
<div class="math">
<p><span class="math">Q = -10 \log_{10} P\ |\ error\ probability = P^2</span></p>
</div><p>So this means:</p>
<img alt="Phred scoring table" class="align-center" src="_images/basecall.jpg" />
<p><em>From: http://res.illumina.com/documents/products/technotes/technote_q-scores.pdf</em></p>
<p>Illumina tends to output sequence results with a Q &gt; 30. So let&#8217;s have a look at what some raw data looks like in terms of Q-scores before and after trimming adapters and low quality reads.</p>
<img alt="Raw vs trimmed alignment" class="align-center" src="_images/rawvstrimmedalign.jpg" />
<p>This is why we do the trimming before attempting to align the reads to the reference genome. Since we are using FastQC, let&#8217;s have a look at some sample data of what <a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good Illumina data looks like</a>.</p>
<p>So, we have come to the end of the background section. Even with all of the great tools and chemistries that have been developed to handle RNA-seq datasets, the old mantra still applies: <em>garbage in; garbage out</em> and <em>with great power comes great responsibility</em>. Take care in analyzing these sorts of data as they typically influence many downstream experiments.</p>
<p><strong>Questions!</strong></p>
</div>
<div class="section" id="basic-analysis-procedure">
<span id="basicworkflow"></span><h2>Basic analysis procedure<a class="headerlink" href="#basic-analysis-procedure" title="Permalink to this headline">¶</a></h2>
<p>Now that we have began to understand the background of RNA-seq technologies, how libraries
are prepared and sequenced, and thought aboutpotential pitfalls during the data analysis process,
let&#8217;s have a look at the basic workflow and some tools that we will use for each step:</p>
<img alt="RNA-seq workflow using SPARTA" class="align-center" src="_images/workflowoutline.jpg" />
<p>Remember that we can have both single- and paired-end reads. Each type of output will require
slightly different tools and procedure. The data that we will be working with is single-end
Illumina reads.</p>
<dl class="docutils">
<dt>Let&#8217;s brainstorm for a minute:</dt>
<dd>If the Illumina sequencing procedure (as seen in the video above) requires specific adapters, what are some ways we could remove them?
What are some potential issues specifically with our reads that could cause misalignments or no alignments at all to a reference genome?
Why don&#8217;t we use a reference transcriptome instead of a genome since RNA-seq is a <em>transcriptional</em> profiling experiment?
What are other genomic features in bacteria that could potentially be identified using RNA-seq data?</dd>
</dl>
</div>
<div class="section" id="trimmomatic">
<span id="id1"></span><h2>Trimmomatic<a class="headerlink" href="#trimmomatic" title="Permalink to this headline">¶</a></h2>
<p>Trimmomatic is a lightweight java application that can remove Illumina adapter sequences and low quality reads. It uses a sliding window to analyze chunks of each read, examining the quality score, minimum read length, if it corresponds to an adapter sequence, etc. Let&#8217;s have a look at the <a class="reference external" href="http://www.usadellab.org/cms/index.php?page=trimmomatic">documentation</a> to see what each option does.</p>
<p>When we run the analysis, you will likely see some output that looks like this:</p>
<div class="highlight-python"><div class="highlight"><pre>TrimmomaticSE: Started with arguments: -threads 4 /mnt/home/john3434/RNAseq/Data/gly7a.fq.gz /mnt/home/john3434/RNAseq/QC/trimmedgly7a.fq.gz ILLUMINACLIP:/opt/software/Trimmomatic/0.32/adapters/TruSeq3-SE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
Using Long Clipping Sequence: &#39;AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA&#39;
Using Long Clipping Sequence: &#39;AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC&#39;
ILLUMINACLIP: Using 0 prefix pairs, 2 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
Quality encoding detected as phred33
Input Reads: 100000 Surviving: 96867 (96.87%) Dropped: 3133 (3.13%)
TrimmomaticSE: Completed successfully
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">It is important to log this output into a text file somewhere and save it (fortunately the software we are going to use will log it for you). You might want this for a report when you&#8217;re finished.</p>
</div>
<p>Let&#8217;s remind ourselves what each command and parameter is doing. Look through the command and discuss with a neighbor what is going on there. If you don&#8217;t remember what each parameter does, have another look at the <a class="reference external" href="http://www.usadellab.org/cms/index.php?page=trimmomatic">documentation</a>.</p>
<p><strong>Let me know if you have questions by placing a red sticky note on your computer.</strong></p>
</div>
<div class="section" id="fastqc">
<span id="id3"></span><h2>FastQC<a class="headerlink" href="#fastqc" title="Permalink to this headline">¶</a></h2>
<p>FastQC is a piece of software that allows us to analyze the quality of our data before proceeding to aligning the reads to the reference genome. Let&#8217;s have a look again at what <a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good Illumina data</a> and <a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">bad Illumina data</a> look like. This will help us determine the quality of our own sequence based on their examples.</p>
<p>The output from FastQC will look like this (with a different file name instead of &#8216;trimmedgly7a.fq.gz&#8217;:</p>
<div class="highlight-python"><div class="highlight"><pre>Started analysis of trimmedgly7a.fq.gz
    Approx 5% complete for trimmedgly7a.fq.gz
    Approx 10% complete for trimmedgly7a.fq.gz
    Approx 15% complete for trimmedgly7a.fq.gz
    Approx 20% complete for trimmedgly7a.fq.gz
    Approx 25% complete for trimmedgly7a.fq.gz
    Approx 30% complete for trimmedgly7a.fq.gz
    Approx 35% complete for trimmedgly7a.fq.gz
    Approx 40% complete for trimmedgly7a.fq.gz
    Approx 45% complete for trimmedgly7a.fq.gz
    Approx 50% complete for trimmedgly7a.fq.gz
    Approx 55% complete for trimmedgly7a.fq.gz
    Approx 60% complete for trimmedgly7a.fq.gz
    Approx 65% complete for trimmedgly7a.fq.gz
    Approx 70% complete for trimmedgly7a.fq.gz
    Approx 75% complete for trimmedgly7a.fq.gz
    Approx 80% complete for trimmedgly7a.fq.gz
    Approx 85% complete for trimmedgly7a.fq.gz
    Approx 90% complete for trimmedgly7a.fq.gz
    Approx 95% complete for trimmedgly7a.fq.gz
    Analysis complete for trimmedgly7a.fq.gz
</pre></div>
</div>
<p>We can open the report file in a browser like FireFox. Here are two different reports <code class="file docutils literal"><span class="pre">report1.html</span></code> and <code class="file docutils literal"><span class="pre">report2.html</span></code> What do we think? Good or bad data?</p>
<p>Please work with a neighbor and discuss the FastQC analysis reports. Put a green sticky note on your computer once you have done this and viewed the results in a browser.</p>
<p><strong>Presentation time!</strong></p>
<p>Please have one person from each treatment group come and present a <em>representative</em> report from each treatment, assessing the results.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Save your report so that we can compile them at the end of the module.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">RNA-seq background information, data analysis procedure, and details of the analysis tools</a><ul>
<li><a class="reference internal" href="#rna-seq-background-information">RNA-seq background information</a></li>
<li><a class="reference internal" href="#basic-analysis-procedure">Basic analysis procedure</a></li>
<li><a class="reference internal" href="#trimmomatic">Trimmomatic</a></li>
<li><a class="reference internal" href="#fastqc">FastQC</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="wishlist.html"
                        title="previous chapter">Functionality wishlist</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/rnaseqbackground.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="wishlist.html" title="Functionality wishlist"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">SPARTA-teaching 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Benjamin K. Johnson.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
  </body>
</html>